argus/scenarios/service_mock.py

```diff
@@ -268,3 +271,30 @@ def x509():
             TO9FwPXIRcR/XrcMDn7slaHtbILM3P4pIbaXUhvel+qLOAMp6k2iDl2Q
             -----END CERTIFICATE-----
             """.strip())
+
+
+class HTTPMetadataServiceApp(BaseServiceApp):
+    """Custom OpenStack http metadata."""
+
+    METADATA = {
+        "keys": [
+            {
+                "name": "argus_cert",
+                "type": "x509",
+                "data": util.get_certificate()
+            }
+        ] + [{
+            "name": "argus_key",
+            "type": "ssh",
+            "data": data
+        } for data in util.get_public_keys()],
+        "admin_pass": "PASsw0r4&!="
```

Comment:
The password is hardcoded, can't you reuse scenario.get_metadata to retrieve it from the config file?